tests/runtests.py

```diff
@@ -142,18 +143,34 @@ def get_test_modules(gis_enabled):
                 yield test_module
 
 
+def get_label_module(label):
+    """Return the top-level module part for a test label."""
+    path = Path(label)
+    if len(path.parts) == 1:
+        # Then interpret the label as a dotted module name.
+        return label.split('.')[0]
+
+    # Otherwise, interpret the label as a path. Check existence first to
+    # provide a better error message than relative_to() if it doesn't exist.
+    if not path.exists():
+        raise RuntimeError(f'test label path does not exist: {label}')
+    path = path.absolute()
```

Comment:
As far as I'm aware `absolute()` is an undocumented API and it will not work for relative paths with dotted segments:
```
./runtests.py ../tests/backends/sqlite/
Testing against Django installed in '/repo/django/django' with up to 8 processes
Found 13 test(s).
Creating test database for alias 'default'...
Cloning test database for alias 'default'...
Cloning test database for alias 'default'...
Cloning test database for alias 'default'...
Cloning test database for alias 'default'...
Cloning test database for alias 'default'...
Cloning test database for alias 'default'...
Cloning test database for alias 'default'...
Cloning test database for alias 'default'...
System check identified no issues (0 silenced).
EE...........
======================================================================
ERROR: backends.sqlite.test_operations (unittest.loader._FailedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/usr/lib/python3.8/unittest/loader.py", line 34, in testFailure
    raise self._exception
ImportError: Failed to import test module: backends.sqlite.test_operations
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/repo/django/tests/backends/sqlite/test_operations.py", line 7, in <module>
    from ..models import Person, Tag
  File "/repo/django/tests/backends/models.py", line 8, in <module>
    class Square(models.Model):
  File "/repo/django/django/db/models/base.py", line 113, in __new__
    raise RuntimeError(
RuntimeError: Model class backends.models.Square doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
```

 using `resolve()` solved this issue for me. Changed.

Reply:
Thanks! I'm wondering now if `os.path.abspath()` should be used for consistency with `RUNTESTS_DIR`, since that's what it's being compared to: https://github.com/django/django/blob/8b4983cfd429e17c8092f4ff775915327effa6fa/tests/runtests.py#L64
It's probably fine for now though.

Reply:
`Path.resolve()` should work the same as `os.path.abspath()`, see [docs](https://docs.python.org/3/library/pathlib.html?highlight=abspath#correspondence-to-tools-in-the-os-module).

Reply:
Yeah, they're comparable. What I was getting at is that I don't think we can be sure they behave the same on all edge cases because they don't share an implementation. You can see some comments to this effect e.g. [here](https://bugs.python.org/issue41357), [here](https://bugs.python.org/issue39090#msg362627), and [here](https://discuss.python.org/t/pathlib-absolute-vs-resolve/2573/3?u=cjerdonek). But like I said above, I think what's there now should be good enough.

Reply:
> But like I said above, I think what's there now should be good enough.

Agreed